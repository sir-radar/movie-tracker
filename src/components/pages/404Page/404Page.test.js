import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Page404 from './404Page';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

jest.useFakeTimers();

afterEach(cleanup)

const resetStatus = jest.fn();

const renderComponent = () => {
  const div = document.createElement("div");
  return render(<Page404/>, div)
}

describe('Page404 Component', () => {

  it("renders without crashing", () => {
    renderComponent()
  })

  it("does not render nav", () => {
    const {queryByTestId} = renderComponent()
    expect(queryByTestId("nav")).toBeNull()
  })

  it("renders page text correctly", () => {
    const { getByText } = renderComponent(resetStatus)
    expect(getByText('404')).not.toBeNull()
    expect(getByText('Page Not Found')).not.toBeNull()
  })


  it("matches snapshot", () => {
    const tree = renderer.create(<Page404/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

})