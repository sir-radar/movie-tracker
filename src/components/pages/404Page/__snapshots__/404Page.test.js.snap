// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 Component matches snapshot 1`] = `
<main
  className="mt-4 p-3 col-10 col-md-10 offset-1 offset-md-1"
>
  <div
    className="row"
  >
    <div
      className="mx-auto text-center"
    >
      <h1>
        404
      </h1>
      <h2>
        Page Not Found
      </h2>
    </div>
  </div>
</main>
`;
